examples/advanced_operations/add_app_campaign.py

```diff
@@ -0,0 +1,306 @@
+    return resource_name
+
+
+def _create_campaign(client, customer_id, budget_resource_name):
+    """Creates an app campaign under the given customer ID.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID str.
+        budget_resource_name: the budget to associate with the campaign
+
+    Returns:
+        A resource_name str for the newly created app campaign.
+    """
+    campaign_service = client.get_service('CampaignService', version='v2')
+    campaign_operation = client.get_type('CampaignOperation', version='v2')
+    campaign = campaign_operation.create
+    campaign.name.value = f'Interplanetary Cruise App #{uuid4()}'
+    campaign.campaign_budget.value = budget_resource_name
+    # Recommendation: Set the campaign to PAUSED when creating it to 
+    # prevent the ads from immediately serving. Set to ENABLED once you've
+    # added targeting and the ads are ready to serve.
+    campaign.status = (client.get_type('CampaignStatusEnum', version='v2')
+        .PAUSED)
+    # All App campaigns have an advertising_channel_type of
+    # MULTI_CHANNEL to reflect the fact that ads from these campaigns are
+    # eligible to appear on multiple channels.
+    campaign.advertising_channel_type = client.get_type(
+        'AdvertisingChannelTypeEnum', version='v2').MULTI_CHANNEL
+    campaign.advertising_channel_sub_type = client.get_type(
+        'AdvertisingChannelSubTypeEnum', version='v2').APP_CAMPAIGN
+    # Define the bidding strategy during campaign creation. An App campaign
+    # cannot use a portfolio bidding strategy.
+    # App campaigns only support the TARGET_CPA bidding strategy.
+    campaign.bidding_strategy_type = (client
+        .get_type('BiddingStrategyTypeEnum', version='v2').TARGET_CPA)
+    # Set the target CPA to $1 / app install.
+    #
+    # campaign_bidding_strategy is a 'oneof' message so setting target_cpa
+    # is mutually exclusive with other bidding strategies such as
+    # manual_cpc, commission, maximize_conversions, etc.
+    # See https://developers.google.com/google-ads/api/reference/rpc
+    # under current version / resources / Campaign
+    campaign.target_cpa.target_cpa_micros.value = 1000000
+    # Set the App Campaign Settings.
+    campaign.app_campaign_setting.app_id.value = (
+        'com.google.android.apps.adwords')
+    campaign.app_campaign_setting.app_store = (client.get_type(
+        'AppCampaignAppStoreEnum', version='v2').GOOGLE_APP_STORE)
+    # Optional fields
+    campaign.start_date.value = (datetime.now() +
+        timedelta(1)).strftime('%Y%m%d')
+    campaign.end_date.value = (datetime.now() +
+        timedelta(365)).strftime('%Y%m%d')
+    # Optimize this campaign for getting new users for your app.
+    campaign.app_campaign_setting.bidding_strategy_goal_type = (client
+        .get_type('AppCampaignBiddingStrategyGoalTypeEnum',
+                  version='v2').OPTIMIZE_INSTALLS_TARGET_INSTALL_COST)
```

Comment:
I don't think you need the outer parens here.

Reply:
it is necessary due to the `.get_type` method call on the subsequent line

Reply:
Will you go over the character limit if you use:

 ```
client.get_type(
    'AppCampaign ...
```

Reply:
yes